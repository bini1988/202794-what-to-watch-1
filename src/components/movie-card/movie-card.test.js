import React from "react";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card";
import card from "../../mocks/movie-card";

it(`MovieCard correctly renders default markup`, () => {
  const tree = renderer.create(
      <MovieCard card={card}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`MovieCard correctly renders complex markup`, () => {
  const tree = renderer.create(
      <MemoryRouter
        initialEntries = {[`/`]}>
        <MovieCard full card={card}>
          <MovieCard.Wrapper main>
            <MovieCard.Header/>
            <MovieCard.Wrapper>
              <MovieCard.Description>
                <MovieCard.PlayButton/>
                <MovieCard.ListButton/>
                <MovieCard.ReviewButton/>
              </MovieCard.Description>
            </MovieCard.Wrapper>
          </MovieCard.Wrapper>
          <MovieCard.InfoWrapper translate>
            <MovieCard.Poster mode="big"/>
            <MovieCard.About/>
          </MovieCard.InfoWrapper>
        </MovieCard>
      </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
