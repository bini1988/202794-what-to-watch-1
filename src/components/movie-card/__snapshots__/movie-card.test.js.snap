// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard correctly renders complex markup 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "background": "#9B7E61",
    }
  }
>
  <div
    className="movie-card__hero"
  >
    <div
      className="movie-card__header"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="War of the Worlds"
          src="https://es31-server.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg"
        />
      </div>
    </div>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          War of the Worlds
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Adventure
          </span>
          <span
            className="movie-card__year"
          >
            2005
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            type="button"
          >
            <svg
              height={19}
              viewBox="0 0 19 19"
              width={19}
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            type="button"
          >
            <svg
              height={20}
              viewBox="0 0 19 20"
              width={19}
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          <a
            className="btn movie-card__button"
            href="/film/1/review"
            onClick={[Function]}
          >
            Add review
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie-card__wrap movie-card__translate-top"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster movie-card__poster--big"
      >
        <img
          alt="War of the Worlds"
          height="327"
          src="https://es31-server.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <nav
          className="movie-nav movie-card__nav"
        >
          <ul
            className="movie-nav__list"
          >
            <li
              className="movie-nav__item movie-nav__item--active"
            >
              <a
                className="movie-nav__link"
                href="#overview"
              >
                Overview
              </a>
            </li>
            <li
              className="movie-nav__item"
            >
              <a
                className="movie-nav__link"
                href="#details"
              >
                Details
              </a>
            </li>
            <li
              className="movie-nav__item"
            >
              <a
                className="movie-nav__link"
                href="#reviews"
              >
                Reviews
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="movie-rating"
        >
          <div
            className="movie-rating__score"
          >
            6.5
          </div>
          <p
            className="movie-rating__meta"
          >
            <span
              className="movie-rating__level"
            >
              Good
            </span>
            <span
              className="movie-rating__count"
            >
              386834 ratings
            </span>
          </p>
        </div>
        <div
          className="movie-card__text"
        >
          As Earth is invaded by alien tripod fighting machines, one family fights for survival.
          <p
            className="movie-card__director"
          >
            <strong>
              Director: Steven Spielberg
            </strong>
          </p>
          <p
            className="movie-card__starring"
          >
            <strong>
              Starring: Tom Cruise, Dakota Fanning, Tim Robbin and other
            </strong>
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`MovieCard correctly renders default markup 1`] = `
<section
  className="movie-card"
  style={
    Object {
      "background": "#9B7E61",
    }
  }
/>
`;
